core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/environment/EnvironmentsController.java

```diff
@@ -36,24 +40,44 @@
     private ConnectionDtoResourceAssembler connectionDtoResourceAssembler;
+    private IEnvironmentDtoService environmentDtoService;
+    private PagedResourcesAssembler<EnvironmentDto> environmentDtoPagedResourcesAssembler;
 
     @Autowired
     EnvironmentsController(EnvironmentService environmentService, ConnectionService connectionService,
-                           EnvironmentDtoResourceAssembler environmentDtoResourceAssembler, ConnectionDtoResourceAssembler connectionDtoResourceAssembler) {
+                           EnvironmentDtoResourceAssembler environmentDtoResourceAssembler, ConnectionDtoResourceAssembler connectionDtoResourceAssembler,
+                           IEnvironmentDtoService environmentDtoService, PagedResourcesAssembler<EnvironmentDto> environmentDtoPagedResourcesAssembler ) {
         this.environmentService = environmentService;
         this.connectionService = connectionService;
         this.environmentDtoResourceAssembler = environmentDtoResourceAssembler;
         this.connectionDtoResourceAssembler = connectionDtoResourceAssembler;
+        this.environmentDtoService = environmentDtoService;
+        this.environmentDtoPagedResourcesAssembler = environmentDtoPagedResourcesAssembler;
     }
 
     @GetMapping("")
     @PreAuthorize("hasPrivilege('ENVIRONMENTS_READ')")
+    public PagedModel<EnvironmentDto> getAll(Pageable pageable) {
+        Page<EnvironmentDto> environmentDtoPage = environmentDtoService.getAll(pageable);
+
+        if (environmentDtoPage.hasContent())
+            return environmentDtoPagedResourcesAssembler.toModel(environmentDtoPage, environmentDtoResourceAssembler::toModel);
+        return (PagedModel<EnvironmentDto>) environmentDtoPagedResourcesAssembler.toEmptyModel(environmentDtoPage, EnvironmentDto.class);
+    }
+
+    @GetMapping("/list")
+    @PreAuthorize("hasPrivilege('ENVIRONMENTS_READ')")
     public HalMultipleEmbeddedResource<EnvironmentDto> getAll() {
-        List<Environment> environments = environmentService.getAll();
-        return new HalMultipleEmbeddedResource<>(
-                environments.stream().filter(distinctByKey(Environment::getName))
-                        .map(environment -> environmentDtoResourceAssembler.toModel(environment))
-                        .collect(Collectors.toList()));
+        Page<EnvironmentDto> environmentDtoPage = environmentDtoService.getAll(Pageable.unpaged());
+        List<EnvironmentDto> environmentDtos = environmentDtoPage.getContent();
+        HalMultipleEmbeddedResource<EnvironmentDto> halMultipleEmbeddedResource = new HalMultipleEmbeddedResource<>();
+        for (EnvironmentDto environmentDto : environmentDtos) {
+            halMultipleEmbeddedResource.embedResource(environmentDto);
+            Link selfLink = linkTo(methodOn(EnvironmentsController.class).getByName(environmentDto.getName()))
+                    .withSelfRel();
+            environmentDto.add(selfLink);
```

Comment:
combine this piece of code using 
environmentDtoPage.getContent()
    .forEach(halMultipleEmbeddedResource::embedResource)